core/src/test/java/org/apache/druid/testing/DeadlockDetectingFailOnTimeout.java

```diff
@@ -194,13 +194,17 @@ private Thread getStuckThread(Thread mainThread)
     Thread stuckThread = null;
     long maxCpuTime = 0;
     for (Thread thread : threadsInGroup) {
-      if (thread.getState() == Thread.State.RUNNABLE) {
+      try {
+        thread.join();
```

Comment:
This doesn't make sense to me. If the thread is stuck and we are trying to `join()` it, we are stuck, too.

Reply:
@leventov yup, we need to SuppressWarnings for this.